source/simulator/stokes_matrix_free.cc

```diff
@@ -2544,6 +2759,14 @@ namespace aspect
       additional_data.mapping_update_flags = (update_values | update_gradients |
                                               update_JxW_values | update_quadrature_points);
 
+      if (sim.mesh_deformation
+          && !sim.mesh_deformation->get_free_surface_boundary_indicators().empty())
+        additional_data.mapping_update_flags_boundary_faces =
+          (update_values  |
```

Comment:
Why do you not need the gradients here, but you do above?

Reply:
Because we don't need it in the matrix-free operator, material values are precomputed and stored.